packages/vite/src/node/optimizer/index.ts

```diff
@@ -603,7 +603,7 @@ export async function runOptimizeDeps(
             js: `import { createRequire } from 'module';const require = createRequire(import.meta.url);`
           }
         : undefined,
-    target: isBuild ? config.build.target || undefined : ESBUILD_MODULES_TARGET,
+    target: config.build.target ?? isBuild ? undefined : ESBUILD_MODULES_TARGET,
```

Comment:
I don't think this is correct. We're setting a hard coded value in dev because that's the baseline browsers Vite supports by default (and also requires it). The `build.target` config should only affect builds per the config name.

Reply:
@bluwy please explain why is that incorrect to ignore configuration in dev mode and respect in production?

with such strategy you are enforcing differences between dev and prod.

i have spent time investigating and fixing the issue, not in my postinstall script i have to "patch" vite to handle source code properly, that exact change i did. it is quite frustrating to see that my effort is just rejected.

Reply:
The **build** config is meant for builds only. In dev, we set a minimum baseline so that core Vite dev server features works. In most cases, it's expected that you use an evergreen browser for dev, and only test legacy browsers with the prod build, so there shouldn't be any inconsistency.

Reply:
more words saying the same do not help.

i see inconsistency here, vite breaks guarantee that `build` will have same functional outcome as `dev`. it is not addressed